ogcapi-stable/ogcapi-common/src/main/java/de/ii/ldproxy/ogcapi/infra/rest/OgcApiExceptionMapper.java

```diff
@@ -0,0 +1,124 @@
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author zahnen
+ */
+@Component
+@Provides
+@Instantiate
+@Provider
+@Produces({MediaType.WILDCARD})
+public class OgcApiExceptionMapper extends LoggingExceptionMapper<Throwable> implements MessageBodyWriter<OgcApiErrorMessage> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OgcApiExceptionMapper.class);
+
+    @Requires
+    Jackson jackson;
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Validate
+    void onStart() {
+        LOGGER.debug("OGC API EXCEPTION WRITER");
+    }
+
+    @Override
+    public Response toResponse(Throwable exception) {
+        String mediaType = getResponseMediaType();
+        if (exception instanceof WebApplicationException) {
+            final Response response = ((WebApplicationException) exception).getResponse();
+            Response.Status.Family family = response.getStatusInfo().getFamily();
+            if (family.equals(Response.Status.Family.CLIENT_ERROR)) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Exception stacktrace:", exception);
+                }
+                return Response.fromResponse(response)
+                        .type(mediaType)
+                        .entity(new OgcApiErrorMessage(response.getStatus(),
+                                response.getStatusInfo().getReasonPhrase(),
+                                exception.getMessage(),
+                                uriInfo.getRequestUri().toString()))
+                        .build();
+            }
+        }
+        long id = logException(exception);
+        int status = 500;
+        return Response.serverError()
+                .type(mediaType)
+                .entity(new OgcApiErrorMessage(status,
+                        Response.Status.fromStatusCode(status).getReasonPhrase(),
+                        String.format("There was an error processing your request, it has been logged. Error ID: %d", id),
+                        uriInfo.getAbsolutePath().toString()))
+                .build();
+    }
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return type.equals(OgcApiErrorMessage.class);
+    }
+
+    @Override
+    public long getSize(OgcApiErrorMessage e, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return -1;
+    }
+
+    @Override
+    public void writeTo(OgcApiErrorMessage e, Class<?> type, Type genericType, Annotation[] annotations,
+                        MediaType mediaType, MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream)
+            throws IOException, WebApplicationException {
+        if (MediaType.TEXT_HTML.equals(mediaType.toString())) {
```

Comment:
JSON and HTML output of exceptions should also be modelled like all other formats, as separate classes that implement FormatExtension.

Reply:
Okay, will update this